File: LL/src/Presentation/ll/src/app/shared/models/Dtos/guild/guildMember.ts
import { GuildRole } from './guildRole';

export interface GuildMember {
  characterId: string;
  name: string;
  level: number;
  role: GuildRole;
  joinedAt: string; // ISO
  isOnline: boolean;
  lastSeenAt?: string | null;
}

File: LL/src/Core/Domain/Models/Guilds/GuildMember.cs
using Domain.Models.Entities.Characters;

namespace Domain.Models.Guilds;
public class GuildMember
{
    public Guid GuildId { get; set; }
    public Guild Guild { get; set; } = null!;
    public Guid CharacterId { get; set; }
    public Character Character { get; set; } = null!;
    public GuildRole Role { get; set; } = GuildRole.Member;
    public DateTimeOffset JoinedAt { get; set; } = DateTimeOffset.UtcNow;
}